linkup-cli/install

```diff
@@ -0,0 +1,265 @@
+            "Accept": "application/vnd.github+json",
+            "X-GitHub-Api-Version": "2022-11-28"
+        }
+    )
+
+    with urllib.request.urlopen(req) as response:
+        return [GithubRelease.from_json(release) for release in json.load(response)]
+
+
+def get_latest_stable_release() -> GithubRelease:
+    req = urllib.request.Request(
+        "https://api.github.com/repos/mentimeter/linkup/releases/latest",
+        headers={
+            "Accept": "application/vnd.github+json",
+            "X-GitHub-Api-Version": "2022-11-28"
+        }
+    )
+
+    with urllib.request.urlopen(req) as response:
+        return GithubRelease.from_json(json.load(response))
+
+
+def download_and_extract(user_os: OS, user_arch: Arch, release: GithubRelease) -> None:
+    print(f"Looking for asset on release {release.tag_name} for {user_os.value}/{user_arch.value}")
+    asset_pattern = re.compile(rf"linkup-.+-{user_arch.value}-{user_os.value}\.tar\.gz$")
+
+    download_url = next(
+        (
+            asset.browser_download_url
+            for asset in release.assets if asset_pattern.match(asset.name)
+        ),
+        None,
+    )
+
+    if not download_url:
+        print("Could not find matching tarball in the release assets.")
+        sys.exit(1)
+
+    print(f"Downloading: {download_url}")
+    local_tar_path = Path("/tmp") / Path(download_url).name
+
+    with urllib.request.urlopen(download_url) as response, open(local_tar_path, "wb") as out_file:
+        shutil.copyfileobj(response, out_file)
+
+    print(f"Decompressing {local_tar_path}")
+    with tarfile.open(local_tar_path, "r:gz") as tar:
+        tar.extractall(path="/tmp")
+
+    LINKUP_BIN_PATH.mkdir(parents=True, exist_ok=True)
+    shutil.move("/tmp/linkup", LINKUP_BIN_PATH / "linkup")
+    os.chmod(LINKUP_BIN_PATH / "linkup", 0o755)
+
+    print(f"Linkup installed at {LINKUP_BIN_PATH / 'linkup'}")
+    local_tar_path.unlink()
+
+
+def setup_path() -> None:
+    if str(LINKUP_BIN_PATH) in os.environ.get("PATH", "").split(":"):
+        return
+
+    print(f"\nTo start using Linkup, add '{LINKUP_BIN_PATH}' to your PATH.")
+
+    shell = Shell.from_str(os.path.basename(os.environ.get("SHELL", "")))
+    if shell is None:
+        return
+
+    print(f"Since you are using {shell.name}, you can run the following to add to your profile:")
+    print(f"\n  {shell.add_to_profile_command()}")
+    print("\nThen restart your shell.");
+
+
+def parse_arguments(args: List[str]) -> Context:
+    parser = argparse.ArgumentParser(description="Install Linkup CLI")
+
+    parser.add_argument(
+        "--channel",
+        choices=["stable", "beta"],
+        default="stable",
+        help="Release channel to use (default: stable)"
+    )
+
+    parsed = parser.parse_args(args)
+    channel = Channel[parsed.channel]
+
+    return Context(channel=channel)
+
+
+def main() -> None:
+    if command_exists("linkup"):
+        print("Linkup is already installed.")
+        sys.exit(0)
```

Comment:
Should it be possible to rerun this script to upgrade or to ensure that the correct dependencies are installed?

Reply:
...or to switch channels between stable/beta

Reply:
Do you think it needs to support it? 🤔 
After installed, it should be able to manage updates via `linkup update`. And The 3.0.0 will support update to change between stable and beta channels (https://github.com/mentimeter/linkup/pull/221)

For the dependencies part, that could be one case where it might be interesting to do it. Or maybe we should look into the `linkup update` to be able to handle dependencies as well? 🤔

Reply:
Nah I think that's fine, maybe just add to the printout to use linkup update instead?